### PR TITLE
Pal 649 policy service redis

### PR DESCRIPTION
**Pull request by pd104923**

### CODE REVIEW

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetResourcePolicyRequestTest.java [hunk 1]
```diff
@@ -32,69 +31,57 @@
...
 
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.core.Is.is;
-import static org.hamcrest.core.IsEqual.equalTo;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
-@RunWith(JUnit4.class)
+@JsonTest
+@TestInstance(Lifecycle.PER_CLASS)
```
Comment (dev930018) [inline policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetResourcePolicyRequestTest.java]: Why specify this?

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetTypePolicyRequestTest.java [hunk 2]
```diff
@@ -18,41 +18,33 @@
...
-import java.util.stream.Collectors;
-import java.util.stream.StreamSupport;
 
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.core.Is.is;
-import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.jupiter.api.Assertions.assertAll;
 
-@RunWith(JUnit4.class)
+@SpringBootTest
```
Comment (dev930018) [inline policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetTypePolicyRequestTest.java]: Does this need to be so heavyweight? Why not use just a standard unittest?

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceHierarchyProxyTest.java [hunk 3]
```diff
@@ -41,19 +38,12 @@
...
 
-import static org.hamcrest.CoreMatchers.equalTo;
-import static org.hamcrest.CoreMatchers.not;
-import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.collection.IsMapContaining.hasEntry;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
 
-@RunWith(JUnit4.class)
+@SpringBootTest
```
Comment (dev930018) [inline policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceHierarchyProxyTest.java]: Ditto

**Review by dev930018 — DISMISSED** [review]
One more comment is - what was our decision on putting tests in new directories? At the moment, it breaks sonarqube's code coverage metric

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetResourcePolicyRequestTest.java (hunk 1 above)
Comment (developer01189998819991197253) [reply policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetResourcePolicyRequestTest.java]: not neccesarily needed so removed

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetTypePolicyRequestTest.java (hunk 2 above)
Comment (developer01189998819991197253) [reply policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/request/SetTypePolicyRequestTest.java]: removed

File: policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceHierarchyProxyTest.java (hunk 3 above)
Comment (developer01189998819991197253) [reply policy-service/src/unit-tests/java/uk/gov/gchq/palisade/service/policy/service/PolicyServiceHierarchyProxyTest.java]: removed

**Review by ac74475 — DISMISSED** [review]

File: policy-service/pom.xml [hunk 4]
```diff
@@ -60,6 +60,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-cache</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-redis</artifactId>
+        </dependency>
```
Comment (dev930018) [inline policy-service/pom.xml:66]: Possibly need to look ad RedisTemplate, @dev-958 ?

**Review by dev930018 — APPROVED** [review]

File: policy-service/pom.xml (hunk 4 above)
Comment (pd104923) [reply policy-service/pom.xml:66]: Yeah, will probably try and have a look at that today.

**Review by developer01189998819991197253 — APPROVED** [review]

**Review by dbz120896 — APPROVED** [review]

**Pull request merged**